sdk/reservations/Azure.ResourceManager.Reservations/tests/Tests/GetCatalogTests.cs

```diff
@@ -0,0 +1,495 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Reservations.Models;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Reservations.Tests
```

Comment:
Why are there only test about `GetCatalog` method here? Do we plan to continue to add other exposed public API's tests in this PR later?

Reply:
Initially I was thinking to keep this PR small, which was why I only added catalog tests. But I can add tests for other operations here as well so that we can have all the tests checked in one single PR.